tools/sof-tplgreader.py

```diff
@@ -201,13 +245,15 @@ def func_getPipeline(tplgObj, tplgName, sdcard_id, sort):
 
     parser.add_argument('filename', type=str, help='tplg file name, multi-tplg file name use "," to split it')
     parser.add_argument('-s', '--sofcard', type=int, help='sofcard id', default=0)
-    parser.add_argument('-f', '--filter', type=str, nargs='+',
-        help='''setup filter, this value is format value,
-string format is 'key':'value','value'
-for example: Get "type" is "capture"
-type: capture
-for example: Get "type" is "capture, both" and "fmt" is S16_LE
-type:capture,both fmt:S16_LE
+    parser.add_argument('-f', '--filter', type=str,
+        help='''setup filter, command line parameter
+string format is 'key':'value','value', the filter
+string support & | and ~ logic operation.
+if only care about key, you can use 'key':'any'.
+Example 1: `-f "type:capture & fmt:S16_LE"`
```

Comment:
While these examples are good, its best to be explicit about it. Something more like your usage examples in the description for your first iteration would be great.

> all pipelines: sof-tplgreader.py tplg-path -f "type:playback,capture"
capture pipelines with PGA: sof-tplgreader.py tplg-path -f "type:capture" "pga:any"
playback pipelines with EQ: sof-tplgreader.py tplg-path -f "type:playback" "eq:any"
all pipelines with PGA: sof-tplgreader.py tplg-path -f "pga:any"
all pipelines with PGA and EQ: sof-tplgreader.py tplg-path -f "pga:any" "eq:any"
>

Reply:
I have added more example in the help messages.